File: prime_or_not.cpp
/*
input an integer number and check whether it is prime r not
*/
#include<iostream>
using namespace std;
int main(){

   int num,i;
   bool isprime = true;

    cout<<"provide a positive number";
    cin>>num;

    if(num<=2){
        isprime = false;
    }else{
        for(int i=2;i<=num/2;i++){
            if(num%i == 0){
                isprime = false;
            }
        }
    }
    if(isprime){
        cout<<"number is prime";
    }else{
        cout<<"number is not a prime";
    }

    return 0;
}

File: prime_or_not.java

import java.util.Scanner;

public class prime_or_not {
    public static void main(String arg[]){
        System.out.println("check weather it is prime or not :");
        Scanner reg=new Scanner(System.in);

        boolean isprime = true;

        int num1=reg.nextInt();

        if(num1<0){
            isprime = false;
        }else{
            for(int i=0;i<num1-1;i++){
                if(num1 % 2 == 0){
                    isprime = false;
                }
            }
        }

        if(isprime){
            System.out.println("entered number is prime");
        }else{
            System.out.println("entered number is not prime");
        }

        




        reg.close();
    }   
}
